frontend/src/features/dsComponents/body/BodyHeader.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import numeral from 'numeral'
+
+import Dataset, { Structure, schemaToColumns, ColumnProperties } from '../../../qri/dataset'
+import Icon from '../../../chrome/Icon'
+import IconLink from '../../../chrome/IconLink'
+
+interface BodyHeaderProps {
+  data: Dataset
+  onToggleExpanded?: () => void
+  showExpand?: boolean
+}
+
+const extractColumnHeaders = (structure: Structure, value: any[]): ColumnProperties[] => {
+  if (!structure || !value) {
+    return []
+  }
+  const schema = structure.schema
+
+  if (!schema) {
+    const firstRow = value && value[0]
+    if (!firstRow) return []
+    return firstRow.map((d: any, i: number) => `field_${i + 1}`)
+  }
+
+  return schemaToColumns(schema)
+}
```

Comment:
move this to `/src/qri/dataset.ts` (near the `ColumnProperties` definition), and import it here